### ISSUE TITLE
[v.2.0.0] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#94937]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/2.0) for the 2.0.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 02/27): work on finalizing the release branch
* Phase 2 (after 02/27): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 02/27/23):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.13.x for 2.0 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Critical fixes to new features introduced in the most recent minor release (e.g. 1.13.x for 2.0 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently  @weiwangmeta, @atalman, @izaitsevfb, @malfet, @osalpekar ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 02/27/23):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.0).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.0
    git checkout release/2.0
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.13' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.0
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.0)

### Versions

2.0.0

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/94934

Criteria Category:
*  Release Only changes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94785

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/94936

Criteria Category:
* Critical fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/94981

Criteria Category:
* Lint fix

**Comment by mingfeima** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94969

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95018

Criteria Category:
* Critical fixes to new features introduced in the most recent minor release (e.g. 1.13.x for 2.0 release). The feature has been posted at https://github.com/pytorch/pytorch/issues/91951

**Comment by XiaobingSuper** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94890

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95054

Criteria Category:
* Critical fixes https://github.com/pytorch/pytorch/issues/95027. CPP backend meet Crash  issue for timm nfnet_l0 .

**Comment by thiagocrepaldi** [comment]
Link to landed mater PR: https://github.com/pytorch/pytorch/pull/94994

Criteria Category:

* Critical fixes for backward compatibility

**Comment by williamwen42** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94940

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95104

Criteria Category:
* Critical fix, per https://github.com/pytorch/pytorch/issues/94914

**Comment by ppwwyyxx** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/93997

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95150

Criteria Category:
* Test/CI fixes. for compatibility with recent numpy

**Comment by davidberard98** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95085

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95242

Criteria Category:
*  Documentation improvements

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94882

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95284

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95247

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95285

Criteria Category:
* Critical fix - torchtriton

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/commit/18d93cdc5dba50633a72363625601f9cf7253162

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95310

Criteria Category:
* Critical CI fix

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):

- https://github.com/pytorch/pytorch/pull/94952
- https://github.com/pytorch/pytorch/pull/95231 
- https://github.com/pytorch/pytorch/pull/95091
-  https://github.com/pytorch/pytorch/pull/94889

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/95317

Criteria Category:
- Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):
- #95078
- #94871

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/95318

Criteria Category:

- Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):
- #95272
- #95093

Link to release branch PR:
 - https://github.com/pytorch/pytorch/pull/95321

Criteria Category:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):
- #95145
- #91071
- #95113

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/95323

Criteria Category:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):

- https://github.com/pytorch/pytorch/pull/94963
- https://github.com/pytorch/pytorch/pull/90772
- https://github.com/pytorch/pytorch/pull/94877

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/95325

Criteria Category:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):

- https://github.com/pytorch/pytorch/pull/94892
- https://github.com/pytorch/pytorch/pull/94923
- https://github.com/pytorch/pytorch/pull/94933
- https://github.com/pytorch/pytorch/pull/94950

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/95327
 
Criteria Category:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by williamwen42** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95088

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95342

Criteria Category:
* Critical fix, per https://github.com/pytorch/pytorch/issues/94914

**Comment by jeejakp12** [comment]
Link to landed master PR (if applicable):

-     https://github.com/pytorch/pytorch/pull/91702

Link to release branch PR:

-     https://github.com/pytorch/pytorch/pull/95363

Criteria Category:

- Special Dispensation - required for other backend support (like Habana)

**Comment by ngimel** [comment]
Link to landed master PR #95339
Link to release branch PR #95383

Criteria: crash fix

**Comment by kulinseth** [comment]
Link to landed master PR (if applicable):
- https://github.com/pytorch/pytorch/pull/95137

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/95388

Criteria Category:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by drisspg** [comment]
Link to landed master PR (if applicable):

https://github.com/pytorch/pytorch/pull/95311
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95397
Criteria Category:

Critical fixes to new features : SDPA failing with torch compile in certain cases.

**Comment by janeyx99** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95241

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95415

Criteria Category:
* Critical fix - silent correctness (fused adamw implementation w/ FSDP had some issue reports, so we should not default to it)

**Comment by XiaobingSuper** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95361

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95439

Criteria Category:
* Critical fix -inductor CPU backend
----

**Comment by bdhirsh** [comment]
Link to landed master PR:

https://github.com/pytorch/pytorch/pull/95101

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95477

Criteria Category:

- Critical fix: memory leak in AOTAutograd

**Comment by kulinseth** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/95480
Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/95495

Criteria Category:
Critical fixes to new features : MPS backend "Beta" phase.

**Comment by janeyx99** [comment]
Link to landed master PR (if applicable):

https://github.com/pytorch/pytorch/pull/95391

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95516

Criteria Category:

Documentation improvements - Document latest changes to optimizer defaults

**Comment by weiwangmeta** [comment]
Link to landed master PR (if applicable):

https://github.com/pytorch/pytorch/pull/95540

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95577

Criteria Category:

Critical fix for triton packaging issue (cuda.h not found)

**Comment by bdhirsh** [comment]
Link to landed master PR:

https://github.com/pytorch/pytorch/pull/95521

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95614

Criteria Category:

Fixes spurious warning when using `torch.compile`

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95094

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95615

Criteria Category:
* Crash fix on ubuntu 20.04

**Comment by kulinseth** [comment]
Link to landed master PR:

https://github.com/pytorch/pytorch/pull/95480

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95625

Criteria Category:
Critical fixes to new features : MPS backend "Beta" phase.

**Comment by frank-wei** [comment]
Link to landed master PR (if applicable):
https://github.com/pytorch/pytorch/pull/94632

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95627

Criteria Category:
We want to reserve the name tensorrt for torch_tensorrt to prevent any confusion

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/94910

Link to release branch PR:
* 

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95652

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95700

Criteria Category:
*Documentation

**Comment by malfet** [comment]
Link to landed master PR:
- https://github.com/pytorch/pytorch/pull/95559

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/95822

Criteria Category:
 - Regressions from previous release / MPS beta

**Comment by malfet** [comment]
Link to landed master PR:
 - https://github.com/pytorch/builder/pull/1327

Link to release branch commit:
 - https://github.com/pytorch/builder/commit/02d50156a3047d2cf94d6527a51d5e62de745df8

Criteria Category:
  - Critical fixes (syncs conda wheel and package dependencies)

**Comment by dllehr-amd** [comment]
Link to landed master PR:

https://github.com/pytorch/pytorch/pull/95793

Link to release branch PR:

https://github.com/pytorch/pytorch/pull/95834

Criteria Category:
Move Pytorch ROCm to use release version of pytorch-triton-rocm whl

**Comment by ngimel** [comment]
Link to master PR
#95833
Link to release branch PR
#95853

Criteria: fix for a critical bug #95794

**Comment by janeyx99** [comment]
Link to master PR
https://github.com/pytorch/pytorch/pull/95820
https://github.com/pytorch/pytorch/pull/95811

Link to release branch PR
https://github.com/pytorch/pytorch/pull/95862

Criteria: fix for a critical bug/silent correctness in flipping the switch for optim

**Comment by kulinseth** [comment]
Link to master PR:
https://github.com/pytorch/pytorch/pull/95769

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95869

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Link to master PR: 
https://github.com/pytorch/pytorch/pull/95762

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95871

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by heidongxianhua** [comment]
Link to landed master PR (if applicable):


https://github.com/pytorch/pytorch/pull/95179
Link to release branch PR:



https://github.com/pytorch/pytorch/pull/95876
Criteria Category:

optimize the warning message

**Comment by jansel** [comment]
Link to master PR
https://github.com/pytorch/pytorch/pull/95842

Link to release branch PR
https://github.com/pytorch/pytorch/pull/95840

Criteria:
Remove broken config flag (see https://github.com/pytorch/pytorch/issues/95775)

**Comment by kulinseth** [comment]
Link to master PR:
https://github.com/pytorch/pytorch/pull/95870, https://github.com/pytorch/pytorch/pull/95817

Link to release branch PR:

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by kulinseth** [comment]
Master PR:
https://github.com/pytorch/pytorch/pull/95905

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/95978

Critical fixes to new features : MPS backend "Beta" phase.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/95990

Criteria Category:
* Critical CI fix, torchtriton integration

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1330

Criteria Category:
* Critical CI fix, torchtriton integration

**Comment by msaroufim** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95802

Link to release branch PR
* https://github.com/pytorch/pytorch/pull/96007

Criteria Category:
* Important documentation update

**Comment by malfet** [comment]
Link to a landed master PR:
 - NA

Link to a release branch commit:
 - https://github.com/pytorch/pytorch/commit/8e05e41dbce1be2d19250a9581bf54acd53dd9f4

Criteria category: Release branch specific changes (use `builder/release-2.0` for binary tests)

**Comment by crcrpar** [comment]
Link to landed master PR (if applicable):
* #95847

Link to release branch PR:
* #96106

Criteria Category:
* Fix fused Adam and AdamW so that they appropriately skip parameter updates when some gradients are invalid, grad scaling factor is not set, but `found_inf` of 1 is set during FP16 AMP training. The wrong behavior broke karpathy/nanoGPT

**Comment by atalman** [comment]
**Important Note!** As of March 6. The only accepted changes are Release-blocking critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

**Comment by snadampal** [comment]
Master PR:
https://github.com/pytorch/pytorch/pull/96141

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/96169

this addressed the issue https://github.com/pytorch/pytorch/issues/96140 which is a regression from PyTorch 1.13.1.

**Comment by 0x6b64** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95072

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/96200

Criteria Category:
* Critical fix for supporting custom process groups.

@atalman

**Comment by ppwwyyxx** [comment]
I wonder if https://github.com/pytorch/pytorch/pull/91517 falls into accepted changes. It should be pretty low risk. Without this PR, pytorch 2.0 will be incompatible with numpy 1.24+.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/95691

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/96450

Criteria Category:
* Critical dependency fix for torch.compile

**Comment by malfet** [comment]
Link to landed master PR:
 - https://github.com/pytorch/pytorch/pull/96231
Link to release branch PR:
 - https://github.com/pytorch/pytorch/pull/96457

Criteria Category:
  - Regression to a new feature (`torchaudio`+`torch.compile` interop)

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/96040

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/96462

Criteria Category:
* Critical regression

----

**Comment by XuehaiPan** [comment]
Hi, I found `torch` 2.0 wheels are available on `https://download.pytorch.org/whl/torch` and [an anaconda package](https://anaconda.org/pytorch/pytorch) is published. But there is no release made for 2.0 on GitHub, PyPI, and https://pytorch.org. This silently causes CI failures in downstream projects without a notification (e.g., [mypy lint failure](https://github.com/metaopt/torchopt/actions/runs/4422442390/jobs/7754297614), [pytest warning failure](https://github.com/metaopt/torchopt/actions/runs/4422442389/jobs/7754441189)). I wonder if the `torch` 2.0 is finalized? If so, we can start to fix the 2.0-specific failures.

**Comment by atalman** [comment]
@XuehaiPan we just finished promoting to PyPI and Github Release page. There are a lot of moving parts in the release, so we start promotion to download.pytorch.org and anaconda 1 day before official release. So that we can upload all the packages and validate them before promoting to pypi. Sorry for confusion it could cause.
